x-pack/plugins/security_solution/public/common/hooks/search_bar/use_init_timerange_url_params.ts

```diff
@@ -21,42 +22,56 @@ import { inputsActions } from '../../store/inputs';
 import { formatDate } from '../../components/super_date_picker';
 import { useInitializeUrlParam } from '../../utils/global_query_string';
 import { URL_PARAM_KEY } from '../use_url_state';
+import { socTrendsId } from '../../store/inputs/constants';
 
 export const useInitTimerangeFromUrlParam = () => {
   const dispatch = useDispatch();
-
+  const isSocTrendsEnabled = useIsExperimentalFeatureEnabled('socTrendsEnabled');
   const onInitialize = useCallback(
     (initialState: UrlInputsModel | null) =>
-      initializeTimerangeFromUrlParam(initialState, dispatch),
-    [dispatch]
+      initializeTimerangeFromUrlParam(initialState, dispatch, isSocTrendsEnabled),
+    [dispatch, isSocTrendsEnabled]
   );
 
   useInitializeUrlParam(URL_PARAM_KEY.timerange, onInitialize);
 };
 
 const initializeTimerangeFromUrlParam = (
   initialState: UrlInputsModel | null,
-  dispatch: Dispatch
+  dispatch: Dispatch,
+  isSocTrendsEnabled: boolean
 ) => {
   if (initialState != null) {
-    const globalId: InputsModelId = 'global';
     const globalLinkTo: LinkTo = { linkTo: get('global.linkTo', initialState) };
     const globalType: TimeRangeKinds = get('global.timerange.kind', initialState);
 
-    const timelineId: InputsModelId = 'timeline';
     const timelineLinkTo: LinkTo = { linkTo: get('timeline.linkTo', initialState) };
     const timelineType: TimeRangeKinds = get('timeline.timerange.kind', initialState);
 
+    const socTrendsLinkTo: LinkTo = { linkTo: get('socTrends.linkTo', initialState) };
```

Comment:
Refactor thought. Instead of having each type reference the other things it's linked t, we just keep a top level linkedToArray. If you want your view linked to the global range, we just add it to the array?

So instead of:

```
timerange=(global:(linkTo:!(socTrends,timeline),timerange:(from:%272022-08-26T04:00:00.000Z%27,kind:absolute,to:%272022-09-15T03:59:59.999Z%27)),socTrends:(linkTo:!(global,timeline),timerange:(from:%272022-08-06T04:00:00.001Z%27,kind:absolute,to:%272022-08-26T04:00:00.000Z%27)),timeline:(linkTo:!(socTrends,global),timerange:(from:%272022-08-26T04:00:00.000Z%27,kind:absolute,to:%272022-09-15T03:59:59.999Z%27)))
```
we do:

```
timerange=(linked:[global, timeline, socTrends], global:(timerange:(from:%272022-08-26T04:00:00.000Z%27,kind:absolute,to:%272022-09-15T03:59:59.999Z%27)),socTrends:(timerange:(from:%272022-08-06T04:00:00.001Z%27,kind:absolute,to:%272022-08-26T04:00:00.000Z%27)),timeline:(timerange:(from:%272022-08-26T04:00:00.000Z%27,kind:absolute,to:%272022-09-15T03:59:59.999Z%27)))
```
and what could be cool about this is, if something is linked to global, we can just drop that timerange from the url and default to using global. So that url would become:

```
timerange=(linked:[global, timeline, socTrends], global:(timerange:(from:%272022-08-26T04:00:00.000Z%27,kind:absolute,to:%272022-09-15T03:59:59.999Z%27))
```
And you just use the exact same thing for `timeline` and `soctrends`?

Reply:
yes absolutely follow up PR... will need to give it more thought though as soctrends is not exactly the same, but the previous time range. i dont _think_ we need to keep it in the url and we can calculate it based off of the global still, but something to consider.